### PR TITLE
add Hydra Attention

### PR DESCRIPTION
**Pull request by zzoneee**
# Motivation
Add Hydra Attention.

# Modification
1.Add Hydra Attention. Now you can choose to use Hydra Attention or Standard Attention in ViT. (Modif "easycv/models/backbones/vision_transformer.py" and "easycv/models/backbones/vit_transformer_dynamic.py")
2.Add unittest for Hydra Attention. (Add "tests/models/backbones/test_hydraAttention.py")
3.Add config file. (Add "configs/classification/imagenet/deit/imagenet_deit_base_hydra_layer12_patch16_224_jpg.py" and "configs/classification/imagenet/deit/imagenet_deit_base_hydra_layer8_patch16_224_jpg.py")

# Accuracy
DeiT-B（Hydra Attention [12 layers]）：
    Official: 76.40
    Ours: 76.67

DeiT-B（Hydra Attention [8 layers]）：
    Official: 79.45
    Ours: 79.444
(For details, see "https://www.yuque.com/u22483603/cgzswd/kcmw4k")

### CODE REVIEW

File: configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# from PIL import Image
+
+_base_ = 'configs/base.py'
```
Comment (tuofeilunhifi) [inline configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py:3]: Reuse the confign of deit

File: tests/models/backbones/test_hydraAttention.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class HydraAttentionTest(unittest.TestCase):
+
+    def setUp(self):
+        print(('Testing %s.%s' % (type(self).__name__, self._testMethodName)))
+
+    def test_hydraAttention(self):
+        model_path = 'http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/hydra_attention/deit_base_patch16_224%20(Hydra%20Attention%20%5B12%20layers%5D).pth'
+        config_path = 'configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py'
+
+        img_norm_cfg = dict(
```
Comment (tuofeilunhifi) [inline tests/models/backbones/test_hydraAttention.py:20]: Why not use the config argument？

File: configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py (hunk 1 above)
Comment (zzoneee) [reply configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py]: done

File: tests/models/backbones/test_hydraAttention.py (hunk 2 above)
Comment (zzoneee) [reply tests/models/backbones/test_hydraAttention.py]: done

File: docs/source/model_zoo_cls.md [hunk 3]
```diff
@@ -2,28 +2,29 @@
...
+| deitiii-large-patch16-192               | [deitiii-large-patch16-192](https://github.com/alibaba/EasyCV/tree/master/configs/classification/imagenet/vit/imagenet_deitiii_large_patch16_192_jpg.py) | 83.902    | 96.296    | 1170            | 10.17    | [model](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/deitiii/imagenet_deitiii_large_patch16_192/deitiii_large.pth) |
+| deit_base_patch16_224 (Hydra Attention [12 layers]) | [deit_base_patch16_224 (Hydra Attention [12 layers])](https://github.com/alibaba/EasyCV/tree/master/configs/classification/imagenet/vit/deit_base_hydraAtt_patch16_224.py) | 76.67    | 92.872    | 338             | 5.73    | [model](http://pai-vision-data-hz.oss-cn-zhangjiakou.aliyuncs.com/EasyCV/modelzoo/classification/hydra_attention/deit_base_patch16_224%20(Hydra%20Attention%20%5B12%20layers%5D).pth) |
```
Comment (wenmengzhou) [inline docs/source/model_zoo_cls.md:27]: hydra attention的精度 相比imagenet_deitiii_base_patch16_192_jpg  降低了？

**Comment by tuofeilunhifi** [comment]
LGTM

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**